### ISSUE TITLE
Bash Utils/Reference Page

### ISSUE THREAD

**Issue by Robadob** [RSE-Sheffield/RSE-Sheffield.github.io#567]
This is something I discussed with Pete a few days ago, as a potential blog post, but it might work better as a permanent reference page on the site, announced with a blog post (similarly could announce if we add to it).

As someone who predominantly works on Windows, but uses Linux, I periodically need to do stuff with bash, which should be trivial, except I can't remember the command structure off hand, and they're not in my recent bash history. Similarly, many of these tools are highly complex with documentation to match, but for 90% of usage there's only a small subset of commands that are required.

Examples:
* `tmux`: create new session, create pane, delete pane, recover session etc.
* `sed`: (and in combination with `find -exec`): Mass perform find-replace across files/directories of files.
* `7z`: extract/add-to archive
* `module`: load, avail, list etc (more limited to hpc)
* `grep`: find things across files (not really one I use, but maybe i should, iirc Pete uses it alot)

Those are the ones that initially come to mind, but it could hopefully grow as a collection of brief reference for common usage of such tools.

Not sure when i'll get to actually drafting it, but if people think it's a good idea I can be harassed to do it at some point when im not busy.

**Comment by bobturneruk** [comment]
Sounds good, @Robadob - thanks for sharing.

**Comment by willfurnass** [comment]
There's already a blog post on tmux (wot I wrote): https://rse.shef.ac.uk/blog/tmux-intro/

I'd argue that would be better to cover uses of `tar` rather than `7z`

Could also include some 'modern' alternatives to grep/find etc e.g. `rg` (ripgrep, which I use lots ever day), `fd`...

**Comment by Robadob** [comment]
> There's already a blog post on tmux (wot I wrote): https://rse.shef.ac.uk/blog/tmux-intro/

Yeah, I was thinking there might have been some suggestions in old newsletters.

> I'd argue that would be better to cover uses of `tar` rather than `7z`

I'm sure alot is up for debate, 7z is all i've ever used. Tar is presumably more likely to be available everywhere.

> Could also include some 'modern' alternatives to grep/find etc e.g. `rg` (ripgrep, which I use lots ever day), `fd`...

This is only going to come about from me drafting stuff, and reviewers suggesting changes. I'll get to it at some point.
